lib/Sema/CodeSynthesisDistributedActor.cpp

```diff
@@ -271,9 +271,14 @@ deriveBodyDistributed_thunk(AbstractFunctionDecl *thunk, void *context) {
       auto recordArgumentDeclRef = UnresolvedDeclRefExpr::createImplicit(
           C, recordArgumentDecl->getName());
 
+      auto argumentName = param->getArgumentName().str();
       auto recordArgArgsList = ArgumentList::forImplicitCallTo(
           recordArgumentDeclRef->getName(),
           {
+            // name:
+           new (C) StringLiteralExpr(argumentName, SourceRange(),
```

Comment:
Note that doing this might back-fire for users of there are multiple empty labels in a declaration i.e. using JSON dictionary could give surprising results depending on a particular JSON implementation.

Reply:
Yeah naively implementing it can definitely backfire, also since there can be `hi(one:one:)` and that'd be wrong in a JSON dictionary. But I think it enables a bunch of nice use-cases when used right so I'm inclined to include it.

I can document some gottchas maybe on the function and we're good here?

Reply:
Are duplicate labels accepted? I don’t think they should be… One suggestion for unlabeled positions could be to use ‘#<index>’ instead like we do in diagnostics.

Reply:
Why wouldn't duplicate labels be accepted? It's bad style but not wrong to declare such methods in Swift after all.

So for unlabeled you're suggesting we should pass `#0`? Hm... maybe... but maybe then I should make some RemoteCallTargetLabel and put the info in there instead? A bit weird to put this into the label string

Reply:
Maybe with a different internal name but just `(one:one:)` would be an error l believe because it’s ambiguous which parameter is referred by `one` in the body.

Yeah, `RemoteCallArgumentLabel` might be a better design. I think we should make it easy to understand how the mistake could be made here as early as possible…

Reply:
yeah diff internal names would be needed; weird but not illegal.

Hm... I'm not so sure the struct helps a lot but I'm not opposed to it 🤔

Reply:
I could safe both internal and external name to help disambiguate maybe…

Reply:
Oh that is interesting... I guess we can offer both yeah...

Reply:
And for situations like `(_:)` or `(a _:)` we’d turn internal label into `#<index>`.

Reply:
oh actually I see what you mean now... internally the synthesis would actually fail for duplicated labels 😱 
Right I need to handle that...

And while at it, introduce the labels info to users... So just that struct with both labels, and implementations can encode them however they want then I guess...

Thanks for noticing

Reply:
Yeah, no worries!

Reply:
impl actually survives it well, we use the internal names.

Adding the struct now...